### PR TITLE
Add space between a number and pm if needed

### PR DESCRIPTION
**Pull request by jaredsohn**
If a user types "4:00pm", make sure it parses as "4:00 pm" instead of "4:00 am".

Also, use an existing moment instead of passing in a date to remove some warnings.

### CODE REVIEW

File: dist/react-datepicker.js [hunk 1]
```diff
@@ -440,6 +440,15 @@ return /******/ (function(modules) { // webpackBootstrap
 
 	    var timeFormats = ["HH:mm:ss a"];
 
+	    var manualDateString = this.state.manualDate;
+
+	    // Add a space between a number and pm if needed
+	    if (manualDateString) {
+	      for (var i = 0; i <= 9; i++) {
+	        manualDateString = manualDateString.replace(new RegExp(i.toString() + "pm", "g"), i.toString() + " pm");
+	      }
```
Comment (dmnd) [inline dist/react-datepicker.js]: Consider using `[0-9]pm` for this

Comment (jonopray) [inline dist/react-datepicker.js]: Yeah, no way you should be looping yourself, use a regex

Comment (jaredsohn) [inline dist/react-datepicker.js]: How do you do a regex replace such that it doesn't replace the number as well? I was hoping I could replace it with "[0-9] pm" but that doesn't work.

Comment (dmnd) [inline dist/react-datepicker.js]: Try this: `"12pm".replace(/([0-9]+)pm/, '$1 pm')`

Comment (jonopray) [inline dist/react-datepicker.js]: Elaborating: you use parentheses `(some match)` (sometimes you have to escape the parens) to capture a match, and then typically insert the match in the replacement using `\1` or `$1`

Comment (jonopray) [inline dist/react-datepicker.js]: Constraining to two numbers: `"12pm".replace(/([0-9]{1,2})pm/, '$1 pm')`

Comment (jaredsohn) [inline dist/react-datepicker.js]: Thanks; I have made a change (but I constrained it to just one number since the input will be both hours and minutes.)

**Comment by jonopray** [comment]
b33881c looks much better

**Pull request merged**